src/coord/src/persistcfg.rs

```diff
@@ -58,7 +59,13 @@ impl TryFrom<String> for PersistStorage {
                     .strip_prefix('/')
                     .unwrap_or_else(|| url.path())
                     .to_string();
-                Ok(PersistStorage::S3(PersistS3Storage { bucket, prefix }))
+                let mut query_params = url.query_pairs().collect::<HashMap<_, _>>();
+                let role_arn = query_params.remove("role_arn").map(|x| x.into_owned());
```

Comment:
Our snowflake precedent starts to fall apart here a bit, they have these as SQL params: https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html#cloud-provider-parameters-cloudproviderparams

Also, I should probably call this "aws_role_arn" or "storage_aws_role_arn" to match.